simapp/app.go

```diff
@@ -263,10 +267,23 @@ func NewSimApp(
 	// not include this key.
 	memKeys := sdk.NewMemoryStoreKeys(capabilitytypes.MemStoreKey, "testingkey")
 
-	// register the streaming service with the BaseApp
-	if err := bApp.SetStreamingService(appOpts, appCodec, keys); err != nil {
-		logger.Error("failed to load state streaming", "err", err)
-		os.Exit(1)
+	// register streaming services
+	streamingCfg := cast.ToStringMap(appOpts.Get(baseapp.StreamingTomlKey))
+	for service := range streamingCfg {
+		pluginKey := fmt.Sprintf("%s.%s.%s", baseapp.StreamingTomlKey, service, baseapp.StreamingABCIPluginTomlKey)
+		pluginName := strings.TrimSpace(cast.ToString(appOpts.Get(pluginKey)))
+		if len(pluginName) > 0 {
+			logLevel := cast.ToString(appOpts.Get(flags.FlagLogLevel))
+			plugin, err := streaming.NewStreamingPlugin(pluginName, logLevel)
+			if err != nil {
+				fmt.Printf("failed to load streaming plugin: %s", err)
+				os.Exit(1)
+			}
+			if err := baseapp.RegisterStreamingPlugin(bApp, appOpts, keys, plugin); err != nil {
+				fmt.Printf("failed to register streaming plugin: %s", err)
+				os.Exit(1)
+			}
+		}
```

Comment:
this should be cleaned up. there can be a helper function that does this

Reply:
moved it into function in `test_helpers.go`